### ISSUE TITLE
Average loss softer & stop training

### ISSUE THREAD

**Issue by lupotto** [AlexeyAB/darknet#845]
Hi @AlexeyAB ,

First of all, thank you for this code/repo, it helped me a lot, concretely training with own dataset. I have two questions:

1. There is some way to modify my loss curve to make it with less noise, either changing my learning rate, cfg file, changing function, etc. 

2. At what iteration do you think that I should stop? My actual avg loss is around 0.7. (attached my current graphic)

![chart](https://user-images.githubusercontent.com/23487343/40190177-2abc3236-59cc-11e8-806b-d18c26f4eebf.jpg)

Thank you.

**Comment by AlexeyAB** [comment]
Lower `subdivision=` (i.e. bigger mini-batch) and lower `learning_rate=` will make loss curve with less noise.

So try to set these params in cfg-file. And stop after 12 000 iterations.
```
steps=10000,11000
scales=.1,.1
```
it will reduce 10 x times the laerning_rate after 10 000 and 11 000 iterations.

* How many images do you use?
* And what params did you change in the cfg-file?

**Comment by lupotto** [comment]
I have 73981 images.

In the cfg file I changed:

batch = 32
subdivision = 8
max_batches = 50200
steps = 40000, 45000
scales = .1,, .1

Also I used your script for find the anchors of my own dataset, 

Thanks for the quick answer!

**Comment by lupotto** [comment]
Thank you Alexey, I will try this parameters. 

Would you recommend me to set backprop to 1, my GPU is a bit old and it's slow training.

**Comment by lupotto** [comment]
Okey perfect, I'll let you know about the results with this changes.

**Comment by lupotto** [comment]
Hi Alexey,

Worked smoothly with your tips.

With this weights I am doing my own testing module. I tried to use this command: `./darknet detector test data/voc.data yolo-voc.cfg yolo-voc.weights -dont_show -ext_output < data/train.txt > result.txt`

It worked, saving the images & detections in results.txt, the thing is that I have 73981 images but when I parse the file appear 54320 images. Do you know why it could be?

Thank you

**Comment by lupotto** [comment]
I solved the problem,

train.txt was created in a bad way.
